velox/functions/prestosql/Sequence.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/vector/ConstantVector.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+class SequenceFunction : public exec::VectorFunction {
+ public:
+  static constexpr int32_t kMaxResultEntries = 10'000;
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    VectorPtr localResult;
+    try {
+      if (args[0]->type() == BIGINT()) {
+        localResult = applyNumber(rows, args, outputType, context);
+      } else {
+        VELOX_NYI(
+            "Unsupported type for argument of 'sequence' function: ",
+            args[0]->type()->toString());
+      }
+    } catch (const std::exception& e) {
+      context.setErrors(rows, std::current_exception());
+      return;
+    }
+    context.moveOrCopyResult(localResult, rows, result);
+  }
+
+ private:
+  static VectorPtr applyNumber(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context) {
+    const auto numRows = rows.size();
+    auto pool = context.pool();
+    vector_size_t numElements = 0;
+
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto startVector = decodedArgs.at(0);
+    auto stopVector = decodedArgs.at(1);
+    DecodedVector* stepVector = nullptr;
+    if (args.size() == 3) {
+      stepVector = decodedArgs.at(2);
+    }
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
```

Comment:
If some rows generated an error, we need to make sure not to process these in the next loop. See usage of context.deselectErrors in Map.cpp

Reply:
@mbasmanova 
So, I've got a question on how "try" should work.
Say we have 100 rows batch and row 5 has incorrect arguments, so checkArguments() throws.
1. Do we need to only return NULL for that row and process the rest?
2. Do we have a way to stop processing rows if there is no "try" outside and such error happened?

Does Map.cpp have all this?